client/src/hooks/useUser.ts

```diff
@@ -0,0 +1,21 @@
+import { useEffect, useState } from 'react'
+import { request } from '../utils/request'
+import { User } from './useUsers'
+
+export function useUser(id: string | undefined): [User | undefined, string, boolean] {
+  const [user, setUser] = useState<User>()
```

Comment:
Doesn't Typescript complain here about setting the initial state as `undefined`?